### PR TITLE
Feature/add db service

### PR DESCRIPTION
**Pull request by slashformotion**
### Added
- database related configuration (fully tested)
- a database "service"
- Tests for the `HTTPServerConfiguration`

### Modified
- The example configuration
- The configuration for the e2e tests

### CODE REVIEW

File: configuration/DatabaseConfiguration_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	assert.Equal(t, 26257, databaseConfiguration.GetPort(), "should be equals")
+}
+
+func TestDatabaseConfigurationGetHost(t *testing.T) {
+	setupViperEnvironment(databaseConfigurationString)
+	databaseConfiguration := configuration.NewDatabaseConfiguration()
+	assert.Equal(t, "e2e-db-1", databaseConfiguration.GetHost())
+}
+
+func TestDatabaseConfigurationDatabaseConfigurationGetUsername(t *testing.T) {
```
Comment (k-y0u) [inline configuration/DatabaseConfiguration_test.go:54]: `TestDatabaseConfigurationDatabaseConfigurationGetUsername` -> `TestDatabaseConfigurationGetUsername`

File: persistence/db/db.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	err = rawDB.Ping()
+	if err != nil {
+		return fmt.Errorf("can't ping the database (%w)", err)
+	}
+	return nil
+}
+
+// Return the database.
+//
+// Panics it the database is nil.
```
Comment (k-y0u) [inline persistence/db/db.go:67]: `Panics it` -> `Panics if`

Ps: What's this message ? :rofl:

File: persistence/db/db.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+)
+
+// the list of models (eg tables for the database) to use in migrations.
+var listOfDatabaseTables = []any{}
+
+// Database instance for the whole app
+var localDB *gorm.DB
+
+// Create the dsn string from the configuration
+func createDsnFromconf() string {
```
Comment (k-y0u) [inline persistence/db/db.go:18]: `createDsnFromconf` -> `createDsnFromConf`

File: persistence/db/db.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+// Create the dsn strings with the provided args
+func createDsn(host, username, password, sslmode, dbname string, port int) string {
+	return fmt.Sprintf("user=%s password=%s host=%s port=%d sslmode=%s dbname=%s",
+		username, password, host, port, sslmode, dbname,
+	)
+}
+
+// Initialize the database with the configuration loaded by verdeter.
+func InitializeDBFromConf() error {
```
Comment (k-y0u) [inline persistence/db/db.go:39]: No test

File: persistence/db/db.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	dsn := createDsnFromconf()
+	return InitializeDBFromDsn(dsn)
+}
+
+// Initialize the database with the database with the dsn string
+func InitializeDBFromDsn(dsn string) error {
```
Comment (k-y0u) [inline persistence/db/db.go:45]: No test

File: persistence/db/db.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+//
+// Panics it the database is nil.
+func GetDB() *gorm.DB {
+	if localDB == nil {
+		panic("the Database was de-allocated")
+	}
+	return localDB
+}
+
+func AutoMigrate() error {
```
Comment (k-y0u) [inline persistence/db/db.go:75]: No test

File: badaas.yml.example [hunk 7]
```diff
@@ -1,3 +1,25 @@
...
+  port: 26257
+  
+  # The sslmode of the connection to the database server.
+  sslmode: disable
+
+  # The username of the account on the database server
+  username: root
+  
+  # The password of the account on the database server
+  password: postres
```
Comment (k-y0u) [inline badaas.yml.example:25]: need "new line line at the end of the file"

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: persistence/db/db.go (hunk 4 above)
Comment (slashformotion) [reply persistence/db/db.go:39]: Not tested for now

File: persistence/db/db.go (hunk 5 above)
Comment (slashformotion) [reply persistence/db/db.go:45]: Not tested for now

**Review by k-y0u — APPROVED** [review]

**Pull request merged**